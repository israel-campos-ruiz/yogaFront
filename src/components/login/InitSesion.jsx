import { useState } from 'react';
import { useMutation } from "@apollo/client";
import { useFormik } from "formik";
import { useContext } from "react";
import { NavLink, useHistory } from "react-router-dom";
import { AUTHENTICATE_USER } from "../../helpers/mutations";
import { getFromLocalStorage, saveLocalStorage } from "../../helpers/storage";
import { AuthContext } from "../../context/AuthContext";
// import Swal from "sweetalert2";
// import { sweetAlertImg } from "../../utils/utils";
import client from "../../apollo/client"

const InitSesion = ({ changeToCreateAccount }) => {
  const [authenticateUser] = useMutation(AUTHENTICATE_USER, {client});
  const [showErrorLogin, setShowErrorLogin] = useState(null)
  const [visibilityPassword, setVisibilityPassword] = useState(false);
  const context = useContext(AuthContext);
  const history = useHistory();
  const initialValues = {
    email: "",
    password: "",
  };

  const onSubmit = async ({ email, password }) => {
     try {
      const { data } = await authenticateUser({
        variables: {
          input: {
            email,
            password,
          },
        },
      });
      saveLocalStorage("auth", data.authenticateUser);
     context?.setAuth(getFromLocalStorage("auth"));
      history.push("/inicio");
     } catch (error) {
       console.log(error.message);
       setShowErrorLogin(error.message)
     }
  };
  const formik = useFormik({
    initialValues,
    onSubmit,
  });
  const handleSee = () => setVisibilityPassword(!visibilityPassword);
  return (
    <>
     
      <form className="form" onSubmit={formik.handleSubmit}>
        <div className="form__container--input">
          <input
            className="form__input"
            placeholder="email"
            name={"email"}
            {...formik.getFieldProps("email")}
          />
        </div>
        <div
          className={`form__container--input ${showErrorLogin && "errors"}`}
        >
          <input
            type={visibilityPassword ? "text" : "password"}
            className={`form__input ${
              formik.touched.password &&
              formik.errors.password &&
              "input__text__error"
            }`}
            name={"password"}
            placeholder="ingresa tu contraseña"
            {...formik.getFieldProps("password")}
          />
          {visibilityPassword ? (
            <i onClick={handleSee} className="fa fa-eye"></i>
          ) : (
            <i onClick={handleSee} className="fa fa-eye-slash"></i>
          )}

    {showErrorLogin && (
          <span className="input__error">{showErrorLogin}</span>
        )}
        </div>
        <button className={`btn--submit--create ${showErrorLogin && "mt"} `} type="submit">
          Iniciar sesion
        </button>
        <p  className="btn--link link" onClick={changeToCreateAccount}>
          ¿No tienes una cuenta? ¡Registrate ahora!
        </p>
        <NavLink className="link" exact to="/inicio">
          Regresa al inicio
        </NavLink>
      </form>
    </>
  );
};

export default InitSesion;
