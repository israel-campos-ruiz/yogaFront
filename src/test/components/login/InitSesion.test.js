import '@testing-library/jest-dom'
import { shallow } from 'enzyme'

import InitSesion from '../../../components/login/InitSesion'
describe('Prueba en el componente iniciar sesion', () => {
    const onSubmit = jest.fn();
    test('Debe de tomar un snapshot del  componente iniciar sesion', () => {
        const wrapper = shallow(<InitSesion/>)
        expect(wrapper).toMatchSnapshot();
    });

    test('No debe de enviar nada', () => {
       const wrapper = shallow(<InitSesion />);
      wrapper.find('form').simulate('submit', {preventDefault:()=>{}})
      expect(onSubmit).not.toHaveBeenCalled()
    })
    
})