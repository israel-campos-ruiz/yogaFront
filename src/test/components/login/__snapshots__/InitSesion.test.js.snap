// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en el componente iniciar sesion Debe de tomar un snapshot del  componente iniciar sesion 1`] = `
<Fragment>
  <form
    className="form"
    onSubmit={[Function]}
  >
    <div
      className="form__container--input"
    >
      <input
        className="form__input"
        name="email"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="email"
        value=""
      />
    </div>
    <div
      className="form__container--input"
    >
      <input
        className="form__input"
        name="password"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="password"
        value=""
      />
    </div>
    <button
      className="btn--submit--create"
      type="submit"
    >
      Iniciar sesion
    </button>
    <p
      className="btn--link link"
    >
      ¿No tienes una cuenta? ¡Registrate ahora!
    </p>
    <NavLink
      className="link"
      exact={true}
      to="/inicio"
    >
      Regresa al inicio
    </NavLink>
  </form>
</Fragment>
`;
